frontend/src/bundles/talent-onboarding/components/profile-step/profile-step.tsx

```diff
@@ -0,0 +1,274 @@
+            fieldState: ControllerFieldState;
+            formState: UseFormStateReturn<ProfileStepDto>;
+        }): React.ReactElement => {
+            return (
+                <>
+                    {employmentTypeOptions.map((option) => (
+                        <Checkbox
+                            {...{
+                                onChange: field.onChange,
+                                onBlur: field.onBlur,
+                                name: field.name,
+                                value: field.value,
+                            }}
+                            key={option.value}
+                            label={option.label}
+                            value={option.value}
+                            onChange={handleCheckboxOnChange(
+                                field,
+                                option.value,
+                            )}
+                        />
+                    ))}
+                </>
+            );
+        },
+        [handleCheckboxOnChange],
+    );
+
+    const handleSliderOnChange = useCallback(
+        (field: ControllerRenderProps<ProfileStepDto, 'experienceYears'>) =>
+            (event: Event): void => {
+                const mouseEvent = event as MouseEvent;
+                const inputElement = mouseEvent.target as HTMLInputElement;
+                const newValue = inputElement.value;
+                const exactExperienceObject = experienceYearsScaled.find(
+                    (experience) =>
+                        experience.scaledValue === Number.parseInt(newValue),
+                );
+                const exactExperienceValue = exactExperienceObject
+                    ? exactExperienceObject.value
+                    : DEFAULT_PAYLOAD_PROFILE_STEP.experienceYears;
+                field.onChange(exactExperienceValue);
+            },
+        [],
+    );
+
+    const renderSlider = useCallback(
+        ({
+            field,
+        }: {
+            field: ControllerRenderProps<ProfileStepDto, 'experienceYears'>;
+            fieldState: ControllerFieldState;
+            formState: UseFormStateReturn<ProfileStepDto>;
+        }): React.ReactElement => {
+            return (
+                <Slider
+                    {...{
+                        onChange: field.onChange,
+                        onBlur: field.onBlur,
+                        name: field.name,
+                        value: field.value,
+                    }}
+                    className={styles.track}
+                    classes={styles}
+                    value={DEFAULT_PAYLOAD_PROFILE_STEP.experienceYears}
+                    marks={experienceYearsSliderMarks}
+                    step={null}
+                    onChange={handleSliderOnChange(field)}
+                />
+            );
+        },
+        [handleSliderOnChange],
+    );
+
+    return (
+        <FormControl className={styles.form}>
```

Comment:
Maybe the parent wrapper for this element should be a `form` element with an `onSubmit`.

Because we intend to use it here in [Talent Sign Up Layout](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/142/files)

And inside layout we do not have a form yet:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/3ec9e65b-84fc-4d2a-8f32-9e5ebc12c845)


That is if we intend to follow with the flow described here:
https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/issues/172#issue-1872246539

Where we run a PATCH for each on-boarding step.

It makes me wonder, how to submit the form though,